### DISCUSSION TITLE
How can we add a global id with preflight.css?

### DISCUSSION THREAD

**Discussion by kfazil** [tailwindlabs/tailwindcss#9887] [Help]
I am working on a chrome extension content script. This extension is going to inject it's on code inside a targating platform. The issue is targating platform has its own base style setup and tailwind base style is conflicting with that. 

For example - 
- Platform has: 
button {
    appearance: auto;
    writing-mode: horizontal-tb !important;
    font-style: ;
    font-variant-ligatures: ;
    font-variant-caps: ;
    font-variant-numeric: ;
    font-variant-east-asian: ;
    font-weight: ;
    font-stretch: ;
    font-size: ;
    font-family: ;
    text-rendering: auto;
    color: buttontext;
    letter-spacing: normal;
    word-spacing: normal;
    line-height: normal;
    text-transform: none;
    text-indent: 0px;
    text-shadow: none;
    display: inline-block;
    text-align: center;
    align-items: flex-start;
    cursor: default;
    box-sizing: border-box;
    background-color: buttonface;
    margin: 0em;
    padding: 1px 6px;
    border-width: 2px;
    border-style: outset;
    border-color: buttonborder;
    border-image: initial;
}

- Tailwind has: 
button,
input,
optgroup,
select,
textarea {
  font-family: inherit; /* 1 */
  font-size: 100%; /* 1 */
  font-weight: inherit; /* 1 */
  line-height: inherit; /* 1 */
  color: inherit; /* 1 */
  margin: 0; /* 2 */
  padding: 0; /* 3 */
}

Notice padding here, its making difference. 

I need to add a id with tailwind css base style. Any Idea how can we achieve this behaviour?

**Comment by wongjn** [comment]
To add extra base styles in Tailwind, use the [`@layer base` directive](https://tailwindcss.com/docs/adding-custom-styles#adding-base-styles).

**Comment by willjohnathan** [comment]
Can you inject your content into a shadow dom? That is typically what I do and achieves the desired result. Although for other reasons, I am currently looking for other solutions as well.
